### PR TITLE
MAINT: Drop support for Python 3.4

### LINKED ISSUE #1865: Drop Python 3.4?
Looks like newer versions of scipy no longer support 3.4. See, e.g. https://travis-ci.org/arokem/dipy/jobs/547695754

### PR DESCRIPTION
**Pull request by arokem**
Closes #1865

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -35,7 +34,7 @@ matrix:
       env:
         - DEPENDS="$DEPENDS scipy"
     # To test minimum dependencies for Python 3.6:
-    - python: 3.4
+    - python: 3.6
```
Comment (arokem) [inline .travis.yml]: I think that this was a typo, now fixed.

**Comment by arokem** [comment]
Looks like this one is now passing and ready for review.

File: .travis.yml [hunk 2]
```diff
@@ -34,11 +33,11 @@ matrix:
...
       env:
         - DEPENDS="$DEPENDS scipy"
-    # To test minimum dependencies for Python 3.6:
-    - python: 3.4
+    # To test minimum dependencies for Python 3.5:
+    - python: 3.5
       env:
         # Check these values against requirements.txt and dipy/info.py
-        - DEPENDS="cython==0.29 numpy==1.8.2 scipy==1.0 nibabel==2.4.0 h5py==2.4.0"
+        - DEPENDS="cython==0.29 numpy==1.9.0 scipy==1.0 nibabel==2.4.0 h5py==2.4.0"
```
Comment (skoudoro) [inline .travis.yml]: Finally, I have one comment: why do you update this Numpy version ? If we really need to update it, can you do it for `requirements.txt` and `dipy/info.py`

Comment (arokem) [reply .travis.yml]: Because there is no wheel for python 3.5/numpy 1.8.2.

Comment (skoudoro) [reply .travis.yml]: make sense, thanks!

**Comment by skoudoro** [comment]
Thank you @arokem

**Pull request merged**